pkg/occlient/occlient.go

```diff
@@ -1105,6 +1113,49 @@ func (c *Client) WaitForBuildToFinish(buildName string) error {
 	return nil
 }
 
+// WaitAndGetDC block and waits until the DeploymentConfig has updated it's annotation
+func (c *Client) WaitAndGetDC(name string, field string, value string, timeout time.Duration) (*appsv1.DeploymentConfig, error) {
+	glog.V(4).Infof("Waiting for DeploymentConfig %s annotation '%s' to update to '%s'", name, field, value)
+
+	w, err := c.appsClient.DeploymentConfigs(c.Namespace).Watch(metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name=%s", name),
+	})
+	defer w.Stop()
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to watch dc")
+	}
+
+	timeoutChannel := time.After(timeout)
+	// Keep trying until we're timed out or got a result or got an error
+	for {
+		select {
+
+		// Timout after X amount of seconds
+		case <-timeoutChannel:
```

Comment:
why not directly use here:
`case <-time.After(timeout)` instead of https://github.com/redhat-developer/odo/pull/848/files#diff-6025e9b620e9321c03c11bf29e5c4dbdR1129 ?

Reply:
I'm pretty sure it'd be endless if it's that case.. defining `time.After(timeout)` before the loop ensures it exits

Reply:
Sounds good it(time.After(timeout)) returns a new channel every-time :)